substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/ProgressReporter.java

```diff
@@ -240,6 +240,10 @@ public void printUnsuccessfulInitializeEnd() {
     public void printInitializeEnd(Collection<String> libraries) {
         stagePrinter.end(getTimer(TimerCollection.Registry.CLASSLIST).getTotalTime() + getTimer(TimerCollection.Registry.SETUP).getTotalTime());
         l().a(" ").doclink("Version info", "#glossary-version-info").a(": '").a(ImageSingletons.lookup(VM.class).version).a("'").println();
+        String javaVersion = System.getProperty("java.runtime.version");
```

Comment:
I think you will need to store the value in a static final field instead, so that it gets baked in the native-image.

See https://github.com/oracle/graal/pull/2961#discussion_r521973500 and https://github.com/oracle/graal/commit/ee25d6c9344b3b2bda0fb0b54968b1944afb105c

Reply:
@zakkak I'm not sure what you're getting at. Could you rephrase? In this patch it should specifically get the JVM version the native image builder runs on. It should never bake into the native image. A run from CI with this has, which I think is expected:

```
========================================================================================================================
GraalVM Native Image: Generating 'javac' (executable)...
========================================================================================================================
[1/7] Initializing...                                                                                    (8.5s @ 0.20GB)
 Version info: 'GraalVM dev Java 17 CE'
 Java version info: '17.0.3+4-jvmci-22.1-b03'
 C compiler: gcc (linux, x86_64, 7.5.0)
 Garbage collector: Serial GC
[...]
```

Reply:
I was mistakenly worried that the `ProgressReporter` would get the wrong `java.runtime.version` property when using a native launcher but apparently that's not the case according to my tests. Both the bash launcher and the native launcher report the correct Java version. Sorry for the noise.

Reply:
OK, thanks for checking!